stdlib/public/Python/PythonLibrary.swift

```diff
@@ -0,0 +1,200 @@
+        "environment variable with the path to the Python Library."
+      )
+    }
+    
+    self.pythonLibraryHandle = pythonLibraryHandle
+    
+    // Check if Python is legacy (Python 2)
+    self.isLegacyPython = dlsym(pythonLibraryHandle, PythonLibrary.pythonLegacySymbolName) != nil
+    
+    if self.isLegacyPython {
+      PythonLibrary.log(message: "Loaded legacy Python library, using legacy symbols...")
+    }
+  }
+  
+  static func getSymbol<T>(name: String, legacyName: String? = nil, signature: T.Type) -> T {
+    var name = name
+    
+    if let legacyName = legacyName, PythonLibrary.shared.isLegacyPython {
+      name = legacyName
+    }
+    
+    let symbol = unsafeBitCast(
+      dlsym(PythonLibrary.shared.pythonLibraryHandle, name),
+      to: signature
+    )
+    
+    return symbol
+  }
+}
+
+/// Methods of `PythonLibrary` required to load the Python library.
+extension PythonLibrary {
+  
+  private static let supportedMajorVersions: [Int] = Array(2...3).reversed()
+  private static let supportedMinorVersions: [Int?] = [nil] + Array(0...30).reversed()
+  
+  private static let libraryPathVersionCharacter: Character = ":"
+  
+  #if canImport(Darwin)
+  private static var libraryNames = ["Python.framework/Versions/:/Python"]
+  private static var libraryPathExtensions = [""]
+  private static var librarySearchPaths = ["", "/usr/local/Frameworks/"]
+  #elseif os(Linux)
+  private static var libraryNames = ["libpython:", "libpython:m"]
+  private static var libraryPathExtensions = [".so"]
+  private static var librarySearchPaths = [""]
+  #endif
+  
+  private static let libraryPaths: [String] = {
+    var libraryPaths: [String] = []
+    
+    for librarySearchPath in librarySearchPaths {
+      for libraryName in libraryNames {
+        for libraryPathExtension in libraryPathExtensions {
+          let libraryPath =
+            librarySearchPath + libraryName + libraryPathExtension
+          libraryPaths.append(libraryPath)
+        }
+      }
+    }
+    
+    return libraryPaths
+  }()
+  
+  private static func loadPythonLibrary(
+    at path: String, majorVersion: Int, minorVersion: Int? = nil) -> UnsafeMutableRawPointer? {
+    
+    var versionString = String(majorVersion)
+    
+    if let minorVersion = minorVersion {
+      versionString += ".\(minorVersion)"
+    }
+    
+    if let requiredPythonVersion = Environment.version.value {
+      let requiredMajorVersion = Int(requiredPythonVersion)
+      
+      if requiredPythonVersion != versionString &&
+        requiredMajorVersion != majorVersion {
+        return nil
+      }
+    }
+    
+    let path = path.split(separator: libraryPathVersionCharacter)
+      .joined(separator: versionString)
+    
+    return loadPythonLibrary(at: path)
+  }
+  
+  private static func loadPythonLibrary(at path: String) -> UnsafeMutableRawPointer? {
+    log(message: "Trying to load library at '\(path)'... ")
+    
```

Comment:
This function has every statement separated by empty lines; it seems in general they can be removed. The same applies to many other functions in this file but this one in particular stands out. It's a great way to organize code but it loses its value when used between every statement.